File: 18Host/TDT4110/PythonWorkspace/ConsoleUtil.py


def cin_I(s: str) -> int:
    while True:
        try:
            return int(input(s))
        except Exception:
            print("Input not an integer. Try again: ")


def cin_F(s: str) -> float:
    while True:
        try:
            return float(input(s))
        except Exception:
            print("Input not a number. Try again: ")

File: 18Host/TDT4120/scripts/ConsoleUtil.jl
function CinI(prompt::String="")::Int
	print(prompt)
	while true
		try
			return parse(Int, readline())
		catch
			println("Input not an integer; Try again:")
		end
	end
end

function CinF(prompt::String="")::Float64
	print(prompt)
	while true
		try
			return parse(Float64, readline())
		catch
			println("Input not a float; Try again:")
		end
	end
end